src/Serilog.Sinks.Elasticsearch/Sinks/ElasticSearch/ElasticSearchTemplateProvider.cs

```diff
@@ -44,97 +43,205 @@ public class ElasticsearchTemplateProvider
                 case AutoRegisterTemplateVersion.ESv5:
                     return GetTemplateESv5(settings, templateMatchString);
```

Comment:
Does this imply supporting v5?

Reply:
Yes for now I left this in and move the deprecation of 5 and 2 into a separate PR as this one is quite big already

Reply:
Sounds like a good plan